cli/foundation-models/system/inference/visual-question-answering/prepare_data.py

```diff
@@ -0,0 +1,169 @@
+import argparse
+import base64
+import json
+import os
+import shutil
+import urllib.request
+import pandas as pd
+from zipfile import ZipFile
+
+
+def download_and_unzip(dataset_parent_dir: str) -> None:
+    """Download image dataset and unzip it.
+
+    :param dataset_parent_dir: dataset parent directory to which dataset will be downloaded
+    :type dataset_parent_dir: str
+    """
+    # Create directory, if it does not exist
+    os.makedirs(dataset_parent_dir, exist_ok=True)
+
+    # download data
+    download_url = "https://cvbp-secondary.z19.web.core.windows.net/datasets/object_detection/odFridgeObjects.zip"
+    print(f"Downloading data from {download_url}")
+
+    # Extract current dataset name from dataset url
+    dataset_name = os.path.basename(download_url).split(".")[0]
+    # Get dataset path for later use
+    dataset_dir = os.path.join(dataset_parent_dir, dataset_name)
+
+    if os.path.exists(dataset_dir):
+        shutil.rmtree(dataset_dir)
+
+    # Get the name of zip file
+    data_file = os.path.join(dataset_parent_dir, f"{dataset_name}.zip")
+
+    # Download data from public url
+    urllib.request.urlretrieve(download_url, filename=data_file)
+
+    # extract files
+    with ZipFile(data_file, "r") as zip:
+        print("extracting files...")
+        zip.extractall(path=dataset_parent_dir)
+        print("done")
+    # delete zip file
+    os.remove(data_file)
+    return dataset_dir
+
+
+def read_image(image_path: str) -> bytes:
+    """Read image from path.
+
+    :param image_path: image path
+    :type image_path: str
+    :return: image in bytes format
+    :rtype: bytes
+    """
+    with open(image_path, "rb") as f:
+        return f.read()
+
+
+def prepare_data_for_online_inference(dataset_dir: str) -> None:
+    """Prepare request json for online inference.
+
+    :param dataset_dir: dataset directory
+    :type dataset_dir: str
+    """
+    sample_image_1 = os.path.join(dataset_dir, "images", "99.jpg")
+    sample_image_2 = os.path.join(dataset_dir, "images", "1.jpg")
+
+    request_json = {
+        "input_data": {
+            "columns": ["image", "text"],
+            "index": [0, 1],
+            "data": [
+                [
+                    base64.encodebytes(read_image(sample_image_1)).decode("utf-8"),
+                    # For BLIP2 append "Answer:" to the below prompt
```

Comment:
Could we take in a model name as parameter and append accordingly.

Reply:
I thought about this too, but then different models could have different prompt conditions... so kept it this way so that we don't have to update for each model..??